src/payment/checkout/payment-form/StripePaymentForm.test.jsx

```diff
@@ -0,0 +1,283 @@
+    mockStripe = mocks.mockStripe();
+    mockElement = mocks.mockElement();
+    mockElements = mocks.mockElements();
+    mockStripe.elements.mockReturnValue(mockElements);
+    mockElements.create.mockReturnValue(mockElement);
+
+    state = {
+      payment: {
+        basket: {
+          loading: true,
+          loaded: false,
+          submitting: false,
+          redirect: false,
+          isBasketProcessing: false,
+          products: [{ sku: '00000' }],
+          enableStripePaymentProcessor: true,
+        },
+        clientSecret: { isClientSecretProcessing: false, clientSecretId: '' },
+      },
+      feedback: { byId: {}, orderedIds: [] },
+      form: {},
+    };
+
+    const wrapper = mount((
+      <IntlProvider locale="en">
+        <AppContext.Provider value={{ authenticatedUser }}>
+          <reactRedux.Provider store={mockStore(state)}>
+            <Elements stripe={mockStripe}>
+              <StripePaymentForm
+                handleSubmit={() => {}}
+                onSubmitPayment={() => {}}
+                onSubmitButtonClick={() => {}}
+              />
+            </Elements>
+          </reactRedux.Provider>
+        </AppContext.Provider>
+      </IntlProvider>
+    ));
+    paymentElement = wrapper.find(PaymentElement);
+    submitButton = wrapper.find(PlaceOrderButton);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('load stripe', () => {
+    it('creates the payment element', () => {
+      expect(mockElements.create).toHaveBeenCalledWith('payment', null);
+      expect(paymentElement.length > 0).toBe(true);
+    });
+  });
+
+  describe('getRequiredFields', () => {
+    it('returns expected required fields', () => {
+      const testFormValues = [
+        {
+          firstName: '',
+          lastName: '',
+          address: '',
+          city: '',
+          country: 'UK',
+          cardExpirationMonth: '',
+          cardExpirationYear: '',
+          optionalField: '',
+        },
+        {
+          firstName: '',
+          lastName: '',
+          address: '',
+          city: '',
+          country: 'CA',
+          cardExpirationMonth: '',
+          cardExpirationYear: '',
+          optionalField: '',
+        },
+        {
+          firstName: '',
+          lastName: '',
+          address: '',
+          city: '',
+          country: 'US',
+          cardExpirationMonth: '',
+          cardExpirationYear: '',
+          optionalField: '',
+        },
+      ];
+
+      testFormValues.forEach((formValues) => {
+        const requiredFields = formValidators.getRequiredFields(formValues);
+        if (formValues.country) {
+          const { optionalField, ...expectedRequiredFields } = formValues;
+          expect(requiredFields).toEqual(expectedRequiredFields);
+        } else {
+          const { optionalField, ...expectedRequiredFields } = formValues;
+          expect(requiredFields).toEqual(expectedRequiredFields);
```

Comment:
This `if...else` appears to execute the same statements for every condition; is this correct? What motivated the addition of these conditions?

Reply:
Just an oversight I forgot to combine. fixed

Reply:
Nice catch! It's interesting that in the `PaymentForm` test, it has the `if formValues.country` but all of examples in the test form values have a country, so my understanding is that the test will never reach the `else`.

Combining makes sense.